src/framework/application.js

```diff
@@ -952,6 +952,9 @@ Object.assign(pc, function () {
                 var asset = new pc.Asset(data.name, data.type, data.file, data.data);
                 asset.id = parseInt(data.id, 10);
                 asset.preload = data.preload ? data.preload : false;
+                // if this is a script asset and has already been embedded in the page then
+                // mark it as loaded
+                asset.loaded = data.type === 'script' && data.data && data.data.loadingType > 0;
```

Comment:
Will script assets always have a value for loadingType?

Reply:
Old scripts won't. But the check should still return false if the value is not there.